tartufo/data/default_regexes.json

```diff
@@ -26,7 +26,7 @@
   "Heroku API Key": "[hH][eE][rR][oO][kK][uU].*[0-9A-F]{8}-[0-9A-F]{4}-[0-9A-F]{4}-[0-9A-F]{4}-[0-9A-F]{12}",
   "MailChimp API Key": "[0-9a-f]{32}-us[0-9]{1,2}",
   "Mailgun API Key": "key-[0-9a-zA-Z]{32}",
-  "Password in URL": "[a-zA-Z]{3,10}://[^/\\s:@]{3,20}:[^/\\s:@]{3,20}@.{1,100}[\"'\\s]",
+  "Password in URL": "[a-zA-Z]{3,10}://[^/\\s:@]{3,40}:[^/\\s:@]{3,40}@[^\"'\\s\\n]+[\"'\\s]",
```

Comment:
I'm staring at the latter part of this expression and trying to work my brain around it, unsuccessfully... ;)

What we used to have was a host+path part of 1-100 characters followed by a quotation mark or whitespace. At the very least I could agree it's not uncommon to see longer URLs (longer than 100 characters)... okay and as I am typing this I see it's now any amount of stuff that's *NOT* the ending delimiter (or a newline), followed by the ending delimiter.

Two comments here. First, you can't see it looking here, but the input blobs get picked apart a line at a time so you'll never find a newline `\n` in the middle of any of the search string. Second, I think you could write this more clearly using a non-greedy operator. I would suggest trying `.+?` instead of `.{1,100}` -- this (with respect to the latter part of the expression) will match any number of characters up until the first quotation mark or whitespace character.

Structurally, now that I'm thinking about it, this still seems problematic:
* If you have quotes, they usually come in pairs -- I guess this logic is implicitly assuming the URL might be quoted (and that we are looking for the close quote that matches the leading quote ahead of the beginning of the URL). If we thought the URL itself might contain quotation marks, this would stop short, but I don't think that will be a problem in practice.
* If your URL is not quoted at all and appears at the end of the line (perhaps `URL=https://user:password@example.com` in a shell script) then neither the original nor the new expression will match.

So what's probably needed is an expression that stops at a quotation mark, whitespace, or the end of the line. For this purpose, I think your expression is better than a non-greedy expression, because it allows us to simply drop the closing delimiter:
```suggestion
  "Password in URL": "[a-zA-Z]{3,10}://[^/\\s:@]{3,40}:[^/\\s:@]{3,40}@[^\"'\\s\\n]+",
```

Reply:
I forgot to remove the unnecessary `\n` when I first posted this, even though I commented on it earlier. I updated the suggestion to remove it.

Reply:
Hey Scott, I see what you mean. In the case that a URL does not currently end in a quote or whitespace we'll miss. I didn't catch that my first pass on this.
I like your suggested change, but before making it I wanted to bring up something. As we search through a line with this pattern, if we get to the @ sign portion of the regex then we have found a URL that matches the important part of the pattern. With that said, what are your thoughts on this?
`[a-zA-Z]{3,10}://[^/\s:@]{3,40}:[^/\s:@]{3,40}@.+`
Does it matter if we report anything past the URL? I still think your suggestion is the best approach, just asking what is the methodology behind these patterns? 

One additional thing to consider is that we are matching templatized strings that don't contain spaces, or strings that contain ENV variables. I think this could be reported separately from bug #375 

Lastly, I'm not sure if writing tests for this data is appropriate for the project as the default_regexes can be changed by the user, but do you think a smoke-test suite should be put together for the data? One that is run on change of data or publishing a release? Here is some example test data.
```
# Patterns that should match:
https://user:password@example.com/foo/bar/gdg
"https://"USER":"password"@example.com/foo/bar/gdg'
URL=https://user:password@example.com
URL = https://user:password@example.com
url: https://user:password@example.com/foo/bar/gdg
url: "https://user:password@example.com/foo/bar/gdg"
url: 'https://user:password@example.com/foo/bar/gdg'

# ENV file matches along with Jinja2/ERB style templating:
url: 'https://${USERNAME}:${PASSWORD}@example.com/foo/bar/gdg'
url: 'https://<%=USERNAME%>:<%=PASSWORD%->@example.com/foo/bar/gdg'
url: 'https://{%USERNAME%}:<%=PASSWORD%->@example.com/foo/bar/gdg'
```

Thanks for the feedback!